test/root_component_test.dart

```diff
@@ -87,15 +86,15 @@ void main() {
         'are invoked during init', () async {
       fixture = await testBed.create();
 
-      verify(mockCredential.handleClientLoad());
-      verify(mockCredential.initClient());
+      verify(mockGoogleApiDart.loadLibrary('client:auth2'));
+      verify(mockGoogleApiDart.initClient(any, any));
```

Comment:
I tried to verify that the correct input has been passed into the function by doing something like:

```
expect(verify(mockGoogleApiDart.initClient(captureAny, any)).captured, InitArgMatcher(initArgs));
```
but here is the error message i got:
```
  Expected: [object Object]
  Actual: NativeJavaScriptObject:<[object Object]>
```

Not sure if this is related to InitArgs having a `@JS()` annotator with it. Maybe we can take a look together if you have time.

Reply:
sounds like the root cause. i think the arg to the googledartapi shouldnt be JS object. implementation should not be important to the caller. let googledartapi convert it into whatever form it needs to be for the js lib.